cli/src/test/java/com/devonfw/tools/ide/io/FileAccessImplTest.java

```diff
@@ -0,0 +1,299 @@
+package com.devonfw.tools.ide.io;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import com.devonfw.tools.ide.context.AbstractIdeContextTest;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.context.IdeTestContextMock;
+
+/**
+ * Test of {@link FileAccessImpl}.
+ */
+public class FileAccessImplTest extends AbstractIdeContextTest {
+
+  private boolean windowsJunctionsAreUsed(IdeContext context, Path dir) {
+
+    Path source = dir.resolve("checkIfWindowsJunctionsAreUsed");
+    Path link = dir.resolve("checkIfWindowsJunctionsAreUsedLink");
+    context.getFileAccess().mkdirs(source);
+    try {
+      Files.createSymbolicLink(link, source);
+    } catch (IOException e) {
+      return context.getSystemInfo().isWindows();
+    }
+    return false;
```

Comment:
This is from your current build on linux:
```
[INFO] Running com.devonfw.tools.ide.io.FileAccessImplTest
11:05:47.217 [main] [C:] - INFO  - c.d.tools.ide.log.IdeSlf4jLogger - Can not check the Test: testWindowsJunctionsCanNotPointToFiles since windows junctions are not used.
11:05:47.220 [main] [C:] - DEBUG - c.d.tools.ide.log.IdeSlf4jLogger - Deleting symbolic link to be re-created at /tmp/junit16793612142448171343/parent/d1/d11/link1
11:05:47.224 [main] [C:] - INFO  - c.d.tools.ide.log.IdeSlf4jLogger - Can not check the Test: testSymlinkAbsoluteAsFallback since windows junctions are not used and fallback from relative to absolute paths as link target is not used.
11:05:47.226 [main] [C:] - DEBUG - c.d.tools.ide.log.IdeSlf4jLogger - Deleting symbolic link to be re-created at /tmp/junit9783774862290736358/parent/d1/d11/link1
11:05:47.231 [main] [C:] - DEBUG - c.d.tools.ide.log.IdeSlf4jLogger - Deleting symbolic link to be re-created at /tmp/junit2359642677952943407/parent/d1/d11/link1
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.020 s -- in com.devonfw.tools.ide.io.FileAccessImplTest
```
You can run your test locally on Windows and see if there is any difference.